nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,170 @@
+            @PathParam("documentId") String documentId) {
+        String query = String.format("Select * From Document where %s = '%s' or %s = '%s'", //
+                NXQL.ECM_UUID, documentId, //
+                NXQL.ECM_PARENTID, documentId);
+
+        return performIndexing(repositoryName, query);
+    }
+
+    /**
+     * Gets the ES indexing status for the given indexing id.
+     *
+     * @param indexingId the indexing id of the launched indexing
+     */
+    @GET
+    @Path("indexing/{indexingId}/status")
+    public Response getIndexingStatus(@PathParam("indexingId") String indexingId) {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        BulkStatus bulkStatus = bulkService.getStatus(indexingId);
+        Map<String, BulkStatus.State> indexingStatus = new HashMap<>();
+        indexingStatus.put("status", bulkStatus.getState());
+
+        return Response.status(SC_OK).entity(indexingStatus).build();
+    }
+
+    /**
+     * Executes an ES flush on document index of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/flush")
+    public void doFlush(@PathParam("repositoryName") String repositoryName) {
+        Framework.doPrivileged(
+                () -> Framework.getService(ElasticSearchAdmin.class).flushRepositoryIndex(repositoryName));
+    }
+
+    /**
+     * Executes an ES optimize on document index of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/optimize")
+    public void doOptimize(@PathParam("repositoryName") String repositoryName) {
+        Framework.doPrivileged(
+                () -> Framework.getService(ElasticSearchAdmin.class).optimizeRepositoryIndex(repositoryName));
+    }
+
+    /**
+     * Executes an ES flush on all documents indexes.
+     */
+    @POST
+    @Path("flush")
+    public void doFlush() {
+        Framework.doPrivileged(() -> Framework.getService(ElasticSearchAdmin.class).flush());
+    }
+
+    /**
+     * Executes an ES optimize on all documents indexes.
+     */
+    @POST
+    @Path("optimize")
+    public void doOptimize() {
+        Framework.doPrivileged(() -> Framework.getService(ElasticSearchAdmin.class).optimize());
+    }
+
+    /**
+     * Performs an ES indexing on documents of the given repository and that match the NXQL query.
+     *
+     * @param repositoryName the repository name on which we launch the indexing, cannot be {@code null}
+     * @param query the NXQL that documents must match to be indexed, can be {@code null} and in this case all documents
+     *            of the given repository will be indexed
+     * @return the http response containing the location of the launched indexing
+     */
+
+    protected Response performIndexing(String repositoryName, String query) {
+        RepositoryService repositoryService = Framework.getService(RepositoryService.class);
+        if (repositoryService.getRepository(repositoryName) == null) {
```

Comment:
This part is still not needed, this is handled by BAF. The only different thing I see is absence of HTTP status as BAF throws an `IllegalArgumentException`. Can you remove it?

Reply:
yes i agree, my idea is to wrappe the http 500 into http 404, the exception message of BAF when there is an unknown repository contains the `toString` method of the command, and i can't based on it to translate it to an 404. plus that i think having an early control in this case saves us to launch the BAF, this is my point of view, But if you think it's really no need i will remove it and the test that control the 404 error.

Reply:
A simpler solution is to improve the BAF code to replace this exception by a NuxeoException one with correct status.
Maybe this change won't need to be backported, TBD with the team.

The early control you're talking about is already implemented in the BAF service and is also early as we don't submit a command if repository doesn't exist.

Reply:
i will create a jira ticket to add this improvements on BAF on 11.1, yes i agree i think no need to backport